### PR TITLE
Reduce file reading boilerplace in tests

### PR DESCRIPTION
**Pull request by konstin**
Add a convenience function around reading files for snapshotting them from the temporary directory.

### CODE REVIEW

**Review by zanieb — APPROVED** [review]
Thanks!

File: scripts/scenarios/templates/lock.mustache [hunk 1]
```diff
@@ -61,7 +61,7 @@ fn {{module_name}}() -> Result<()> {
     );
 
     {{#expected.satisfiable}}
-    let lock = fs_err::read_to_string(context.temp_dir.join("uv.lock"))?;
+    let lock = context.read("uv.lock")();
```
Comment (fasterthanlime) [inline scripts/scenarios/templates/lock.mustache:64]: this seems wrong, I'm drive-by fixing it in #8093 cc @konstin
